java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -2143,4 +2143,63 @@ public void pfmerge(BaseClient client) {
                         ExecutionException.class, () -> client.pfmerge(key1, new String[] {"foo"}).get());
         assertTrue(executionException.getCause() instanceof RequestException);
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void objectEncoding(BaseClient client) {
+        String nonExistingKey = UUID.randomUUID().toString();
+        String stringRawKey = UUID.randomUUID().toString();
+        String stringIntKey = UUID.randomUUID().toString();
+        String stringEmbstrKey = UUID.randomUUID().toString();
+        String listListpackKey = UUID.randomUUID().toString();
+        String setIntsetKey = UUID.randomUUID().toString();
+        String setHashtableKey = UUID.randomUUID().toString();
+        String setListpackKey = UUID.randomUUID().toString();
+        String hashHashtableKey = UUID.randomUUID().toString();
+        String hashListpackKey = UUID.randomUUID().toString();
+        String zsetSkiplistKey = UUID.randomUUID().toString();
+        String zsetListpackKey = UUID.randomUUID().toString();
+        String streamKey = UUID.randomUUID().toString();
+
+        assertEquals(
+                OK,
+                client
+                        .set(stringRawKey, "a really loooooooooooooooooooooooooooooooooooooooong value")
+                        .get());
+        assertEquals(OK, client.set(stringIntKey, "2").get());
+        assertEquals(OK, client.set(stringEmbstrKey, "value").get());
+        assertEquals(1, client.lpush(listListpackKey, new String[] {"1"}).get());
+        // The default value of set-max-intset-entries is 512
+        for (Integer i = 0; i <= 512; i++) {
+            assertEquals(1, client.sadd(setHashtableKey, new String[] {i.toString()}).get());
+        }
+        assertEquals(1, client.sadd(setIntsetKey, new String[] {"1"}).get());
+        assertEquals(1, client.sadd(setListpackKey, new String[] {"foo"}).get());
+        // The default value of hash-max-listpack-entries is 512
+        for (Integer i = 0; i <= 512; i++) {
+            assertEquals(1, client.hset(hashHashtableKey, Map.of(i.toString(), "2")).get());
+        }
+        assertEquals(1, client.hset(hashListpackKey, Map.of("1", "2")).get());
+        // The default value of zset-max-listpack-entries is 128
+        for (Integer i = 0; i <= 128; i++) {
+            assertEquals(1, client.zadd(zsetSkiplistKey, Map.of(i.toString(), 2d)).get());
+        }
+        assertEquals(1, client.zadd(zsetListpackKey, Map.of("1", 2d)).get());
+        assertNotNull(client.xadd(streamKey, Map.of("field", "value")));
+
+        assertNull(client.objectEncoding(nonExistingKey).get());
```

Comment:
would it be better to split this into multiple tests?  
like:
* `objectEncoding_returns_raw`
* `objectEncoding_returns_int`
* ...
That way it would all these tests instead of failing on the first test (and not really knowing how the rest of the result are impacted).